pxr/usd/usd/prim.h

```diff
@@ -2379,8 +2312,138 @@ class UsdPrimSiblingIterator {
     Usd_PrimFlagsPredicate _predicate;
 };
 
-// Typedef iterator range.
-typedef boost::iterator_range<UsdPrimSiblingIterator> UsdPrimSiblingRange;
+/// Forward iterator range of sibling ::UsdPrim s.  This range type contains a
+/// pair of UsdPrimSiblingIterator s, denoting a half-open range of UsdPrim
+/// siblings.  It provides a subset of container-like API, such as begin(),
+/// end(), front(), empty(), etc.
+class UsdPrimSiblingRange {
+public:
+    /// Iterator type.
+    typedef UsdPrimSiblingIterator iterator;
+    /// Const iterator type.
+    typedef UsdPrimSiblingIterator const_iterator;
+    /// Iterator difference type.
+    typedef std::ptrdiff_t difference_type;
+    /// Iterator value_type.
+    typedef iterator::value_type value_type;
+    /// Iterator reference_type.
+    typedef iterator::reference reference;
+
+    UsdPrimSiblingRange() = default;
+    UsdPrimSiblingRange(const UsdPrimSiblingRange&) = default;
+    UsdPrimSiblingRange(UsdPrimSiblingRange&&) = default;
+    UsdPrimSiblingRange& operator=(const UsdPrimSiblingRange&) = default;
+    UsdPrimSiblingRange& operator=(UsdPrimSiblingRange&&) = default;
+
+    /// Construct with a pair of iterators.
+    UsdPrimSiblingRange(UsdPrimSiblingIterator begin,
+                        UsdPrimSiblingIterator end) : _begin(begin),
+                                                      _end(end) {}
+
```

Comment:
I'm assuming these templated constructors and assignment operators are to perfectly match what boost::iterator_range provides?
I feel like these are overkill given that I would never expect anyone to construct a sibling range without calling the Get*Children functions that return them (and it would be honestly dangerous to do so since sibling iteration is a proxy). How do you feel about just leaving these out?